pkg/k8shandler/certificates_test.go

```diff
@@ -0,0 +1,393 @@
+
+	var (
+		cluster = &loggingv1.ClusterLogging{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "instance",
+				Namespace: constants.OpenshiftNS,
+			},
+			Spec: loggingv1.ClusterLoggingSpec{
+				ManagementState: loggingv1.ManagementStateManaged,
+				Collection: &loggingv1.CollectionSpec{
+					Logs: loggingv1.LogCollectionSpec{
+						Type:        loggingv1.LogCollectionTypeFluentd,
+						FluentdSpec: loggingv1.FluentdSpec{},
+					},
+				},
+			},
+		}
+		masterCASecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      constants.MasterCASecretName,
+				Namespace: constants.OpenshiftNS,
+			},
+		}
+		fluentdSecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      constants.FluentdName,
+				Namespace: constants.OpenshiftNS,
+			},
+		}
+
+		elasticsearchSecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      constants.ElasticsearchName,
+				Namespace: constants.OpenshiftNS,
+			},
+		}
+
+		kibanaSecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      constants.KibanaName,
+				Namespace: constants.OpenshiftNS,
+			},
+		}
+
+		kibanaProxySecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      constants.KibanaProxyName,
+				Namespace: constants.OpenshiftNS,
+			},
+		}
+
+		curatorSecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      constants.CuratorName,
+				Namespace: constants.OpenshiftNS,
+			},
+		}
+	)
+
+	Describe("Certificates", func() {
+		var (
+			client         client.Client
+			clusterRequest *ClusterLoggingRequest
+		)
+
+		BeforeSuite(func() {
+			client = fake.NewFakeClient(
+				cluster,
+				masterCASecret,
+				fluentdSecret,
+				elasticsearchSecret,
+				kibanaSecret,
+				kibanaProxySecret,
+				curatorSecret,
+			)
+			clusterRequest = &ClusterLoggingRequest{
+				Client:  client,
+				Cluster: cluster,
+			}
+			if err := os.RemoveAll(utils.GetWorkingDir()); err != nil {
+				Fail(fmt.Sprintf("%v", err))
+			}
+			os.Setenv("SCRIPTS_DIR", scriptsDir)
+			_ = clusterRequest.CreateOrUpdateCertificates()
+		})
+
+		It("should generate a self-signed CA", func() {
+			secret := &corev1.Secret{}
+			key := types.NamespacedName{Name: constants.MasterCASecretName, Namespace: constants.OpenshiftNS}
+			Expect(client.Get(context.TODO(), key, secret)).Should(Succeed())
+
+			Expect(secret).Should(ContainKeys("ca.key", "ca.crt", "ca.db", "ca.serial.txt"))
+			Expect(secret).Should(SucceedVerifyX509("ca.crt", "ca.crt", "openshift-cluster-logging-signer", nil, nil, nil))
+		})
+
+		It("skip re-generating secrets during restart", func() {
+			secret := &corev1.Secret{}
+			key := types.NamespacedName{Name: constants.MasterCASecretName, Namespace: constants.OpenshiftNS}
+			Expect(client.Get(context.TODO(), key, secret)).Should(Succeed())
+
+			Expect(os.RemoveAll(utils.GetWorkingDir())).To(Succeed())
+			clusterRequest.CreateOrUpdateCertificates()
```

Comment:
a lint error here
`Error return value of `clusterRequest.CreateOrUpdateCertificates` is not checked (errcheck)`